notebook/nbextensions.py

```diff
@@ -299,118 +387,287 @@ def _config_file_name_default(self):
         # We're using a dict as a set - updating with None removes the key
-        cm.update(self.section, {"load_extensions": {name: None}})
-
+        cm.update(section, {"load_extensions": {require: self._toggle_value}})
+    
+    def toggle_nbextension_python(self, package):
+        m, nbexts = _get_nbextension_metadata(package)
+        for nbext in nbexts:
+            section = nbext['section']
+            require = nbext['require']
+            self._toggle_nbextension(section, require)
+
+    def toggle_nbextension(self, require):
+        self._toggle_nbextension(self.section, require)
+        
     def start(self):
+
         if not self.extra_args:
-            sys.exit('No extensions specified')
+            sys.exit('Please specify an nbextension/package to enable or disable')
         elif len(self.extra_args) > 1:
-            sys.exit('Please specify one extension at a time')
+            sys.exit('Please specify one nbextension/package at a time')
+        if self.python:
+            self.toggle_nbextension_python(self.extra_args[0])
+        else:
+            self.toggle_nbextension(self.extra_args[0])
+
 
-        self.disable_nbextension(self.extra_args[0])
+class EnableNBExtensionApp(ToggleNBExtensionApp):
+
+    name = "jupyter nbextension enable"
+    description = "Enable an nbextension using frontend configuration files."
+    _toggle_value = True
+
+
+class DisableNBExtensionApp(ToggleNBExtensionApp):
+    
+    name = "jupyter nbextension disable"
+    description = "Disable an nbextension using frontend configuration files."
+    _toggle_value = None
+
+
+class ListNBExtensionsApp(BaseNBExtensionApp):
+    
+    name = "jupyter nbextension list"
+    version = __version__
+    description = "List all nbextensions known by the configuration system"
+    
+    def list_nbextensions(self):
+        config_dirs = [os.path.join(p, 'nbconfig') for p in jupyter_config_path()]
+        for config_dir in config_dirs:
+            self.log.info('config dir: {}'.format(config_dir))
+            cm = BaseJSONConfigManager(parent=self, config_dir=config_dir)
+            for section in ['common', 'notebook', 'tree', 'edit', 'terminal']:
+                data = cm.get(section)
+                if 'load_extensions' in data:
+                    self.log.info('  {} section'.format(section))
+                    
+                    load_extensions = data['load_extensions']
+                    for x in load_extensions:
+                        self.log.info('   {1} {0}'.format(x, GREEN_ENABLED if load_extensions[x] else RED_DISABLED))
+    
+    def start(self):
+        self.list_nbextensions()
+
+
+_examples = """
+jupyter nbextension list                          # list all configured nbextensions
+jupyter nbextension install --py <packagename>    # install an nbextension from a Python package
+jupyter nbextension enable --py <packagename>     # enable all nbextensions in a Python package
```

Comment:
Yeah, so if a --py package has two extensions, a.js and b.js which both install to an nbextension folder `c/`, you could enable one of them, a for example, by running `jupyter nbextension enable c/a`